content/automate/ManageIQ/Transformation/Ansible.class/__methods__/checkplaybookasaservice.rb

```diff
@@ -0,0 +1,68 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Ansible
+        class CheckPlaybookAsAService
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def set_retry(message, interval = '1.minutes')
+            @handle.log(:info, "#{message}. Retrying.")
+            @handle.root['ae_result'] = 'retry'
+            @handle.root['ae_retry_interval'] = interval
+          end
+
+          def job_status(job)
+            { :job_id => job.id, :job_status => job.status }
+          end
+
+          def set_ansible_job(task, service_request, transformation_hook)
+            job = nil
+            service_request_task = service_request.miq_request_tasks.first
+            unless service_request_task.nil?
+              stack = service_request_task.destination.service_resources.first
+              job = stack.resource unless stack.nil?
```

Comment:
`stack` is already the ansible job instance. There is no `stack.resource` method.